addons/mrp/migrations/10.0.2.0/openupgrade_analysis_work.txt

```diff
@@ -0,0 +1,424 @@
+mrp          / mrp.workorder            / final_lot_id (many2one)       : NEW relation: stock.production.lot
+mrp          / mrp.workorder            / message_follower_ids (one2many): NEW relation: mail.followers
+mrp          / mrp.workorder            / message_ids (one2many)        : NEW relation: mail.message
+mrp          / mrp.workorder            / message_last_post (datetime)  : NEW
+mrp          / mrp.workorder            / move_lot_ids (one2many)       : NEW relation: stock.move.lots
+mrp          / mrp.workorder            / move_raw_ids (one2many)       : NEW relation: stock.move
+mrp          / mrp.workorder            / name (char)                   : NEW required: required
+# NOTHING TO DO
+
+mrp          / mrp.workorder            / next_work_order_id (many2one) : NEW relation: mrp.workorder
+# DONE: Ordered the workorders of a same mrp_production based on the v9 'sequence' field and the ids.
+
+mrp          / mrp.workorder            / operation_id (many2one)       : NEW relation: mrp.routing.workcenter
+mrp          / mrp.workorder            / production_id (many2one)      : NEW relation: mrp.production, required: required
+mrp          / mrp.workorder            / qty_produced (float)          : NEW
+mrp          / mrp.workorder            / qty_producing (float)         : NEW
+mrp          / mrp.workorder            / scrap_ids (one2many)          : NEW relation: stock.scrap
+mrp          / mrp.workorder            / state (selection)             : NEW selection_keys: ['cancel', 'done', 'pending', 'progress', 'ready']
+mrp          / mrp.workorder            / time_ids (one2many)           : NEW relation: mrp.workcenter.productivity
+mrp          / mrp.workorder            / workcenter_id (many2one)      : NEW relation: mrp.workcenter, required: required
+# NOTHING TO DO
+
+mrp          / procurement.order        / property_ids (many2many)      : DEL relation: mrp.property
+# NOTHING TO DO
+
+
+mrp          / stock.move               / consumed_for (many2one)       : DEL relation: stock.move
+# DONE: Pre-migration: Renamed to None and copied this column. That will be used in post-migration to known which moves are consumed for.
+
+mrp          / stock.move               / active_move_lot_ids (one2many): NEW relation: stock.move.lots
+mrp          / stock.move               / bom_line_id (many2one)        : NEW relation: mrp.bom.line
+mrp          / stock.move               / consume_unbuild_id (many2one) : NEW relation: mrp.unbuild
+mrp          / stock.move               / move_lot_ids (one2many)       : NEW relation: stock.move.lots
+mrp          / stock.move               / operation_id (many2one)       : NEW relation: mrp.routing.workcenter
+mrp          / stock.move               / quantity_done_store (float)   : NEW
+mrp          / stock.move               / unbuild_id (many2one)         : NEW relation: mrp.unbuild
+mrp          / stock.move               / unit_factor (float)           : NEW
+# NOTHING TO DO
```

Comment:
There is *much* to do here. If this isn't filled up, no manufacturing orders that were brought over will consume any raw materials, because the code in mrp.workorder record_production method won't have a unit_factor, and thus it skips raising consumed qty (quantity_done) when the workorder is done. At least the workorder_id filling is properly done in populate_stock_move_workorder_id, but without unit_factor filled, that isn't useful alone.

If unit_factor isn't calculated for it and filled in, the result is the following for MO's that were created in earlier Odoo version (and weren't in done state):

 - We record production, to keep it simple lets say it had nothing produced before and we produce it all now
 - record_production() on the last workorder (as stock.move workorder_id IS calculated to point at the last workorder of the production it's a raw material to consume for) will check all stock.moves for that production (via raw_move_ids one2many, the reverse of stock.move workorder_id) unit_factor, and raise quantity_done accordingly. quantity_done stays 0 because unit_factor is not set.
- The production is finished and user clicks "Mark as Done"
- mrp module action_done override is called and the production moves are funneled through move_validate() instead of main action_done()
- move_validate() checks what quantity_done is, and because it has remained zero due to missing unit_factor, it cancels the move instead
- The raw material isn't consumed, the reservation is released by stock.move action_cancel, raw material moves are all in cancel state now
- All raw material stock is completely wrong (the quantity that was reserved for production is released back to quantity on hand), because they were obviously consumed in real life production, but Odoo has cancelled the moves instead.

Reply:
fill <code>unit_factor</code> with what?

Reply:
With the unit factor as it is filled upon new workorder generation for fresh Manufacturing Orders..

I fixed it up for my needs after migration with the following odoo shell script:
```
oves_to_fix = env['stock.move'].search([('raw_material_production_id','!=',False),('state','not in',['cancel','done']),('unit_factor','=',False)])
for move in moves_to_fix:
    move.unit_factor = move.product_uom_qty / ((move.raw_material_production_id.product_qty - move.raw_material_production_id.qty_produced) or 1.0)
```

Something like that needs doing during migration. Maybe should bother with cancel or done moves as well, in case they get reset at some point, but the workflow used in my instance didn't have a need for it. If it is done for cancel/done moves as well, it'll take a significant amount of time if done via odoo methods; if with pure SQL, should be very fast (but potentially different rounding that needs to be cared for maybe?).

If this is not done, all raw materials used by already generated (prior to migration) MO-s would go to cancel after MO is marked as done, completely breaking stock quantities.